forest/daemon/src/daemon.rs

```diff
@@ -314,18 +318,24 @@ pub(super) async fn start(config: Config, detached: bool) -> anyhow::Result<Db>
 
     let config = maybe_fetch_snapshot(should_fetch_snapshot, config).await?;
 
-    select! {
+    tokio::select! {
         () = sync_from_snapshot(&config, &state_manager).fuse() => {},
-        _ = ctrlc_oneshot => {
-            // Cancel all async services
+        _ = tokio::signal::ctrl_c() => {
+            services.shutdown().await;
+            return Ok(db);
+        },
```

Comment:
Just to make sure, this way of handling ctrl-c still maintains the same behavior of attempting a soft shutdown on the first ctrl-c and then doing a hard shutdown on subsequent ctrl-c's?

Reply:
I don't think so, I will confirm it to you. Do you think it's still needed?

Reply:
Not sure. But if we change the behavior then we should make a note about it.

Reply:
Ok, to me we don't really need anymore to handle this hard exit. Forest should be able to gracefully shutdown now.

Reply:
Alright. Let's make a note of it in the changelog.